File: STPAR/stpar.pl
#!/usr/bin/perl
#
while (1)
{
	$len = <STDIN>;
	if ( $len == 0 )
	{
		last;
	}
	$line = <STDIN> ;
	chomp($line);
	@vals = split  ( /\s+/ , $line ) ;
	@stack = ();
	push @stack , 9999;
	$num = @vals;
	$curr = 1;
	@res = ();
	$i=0;	
	$flag = 0 ;
	while ($curr <=$num )
	{
		#print " i $i vals i $vals[$i] curr $curr \n";
		if ( $vals[$i] != $curr )
		{
			if ( $stack[-1] == $curr )
			{
				push @res , $stack[-1] ;
				$pval = pop @stack;
				#print "poping $pval\n";
				$curr++;
				#print "resu @res \n";
			}
			else 
			{
				while ( $vals[$i] != $curr and  $i<$num )
				{
					if ( $vals[$i] < $stack[-1] )
					{

						#print "pushing $vals[$i]\n";
						push @stack , $vals[$i] ;
						$i++;
					}
					else
					{
						#got an error here 
						print "no\n";
						$flag = 1 ;
						last;
					}
				}
				push @res , $curr;
				#print "res @res\n";
				$i++;
				$curr++;
			}
		}
		else
		{
			push @res , $vals[$i];
			#print "Res @res \n";
			$curr ++;
			$i++;
		}
		if ( $flag == 1 )
		{
			last ;
		}

	}
	if ( $flag == 0 )
	{
		print "yes\n";
	}
}
			
				

		

File: STPAR/stpar.py
#!/usr/bin/python

while True :
	length = int(raw_input())
	if length == 0 :
		break
	curr = 1 
	stack_top = 2000 
	order = raw_input().split(' ')
	flag = 0
	for i in range(0,length) :
		if int (order[i]) > curr :
			#check stack top and push if less than that 
			if stack_top > int (order[i]) :
				stack_top = int (order[i])
				#print "pusing " , stack_top
			else :
				print "no"
				flag = 1 
				break
		elif int (order[i]) == curr :
			curr+=1 

	if flag != 1 : 
		print "yes"
	
